### PR TITLE
Refactor: move compare-features script test out of linter

### PR DESCRIPTION
**Pull request by bershanskiy**
This PR moves a unit test out of [linter](https://github.com/mdn/browser-compat-data/blob/master/test/lint.js#L161) and into a separate mocha test file similar to [`utils.test.js`](https://github.com/mdn/browser-compat-data/blob/master/test/utils.test.js). This unit test is still called by `npm run test`, since [`test`](https://github.com/mdn/browser-compat-data/blob/master/package.json#L52) script includes `npm run mocha`.

For reviewer convenience, I split it up into two commits: the actual change is in the first commit and the second commit is just `npm run fix`.

### Before:
```
$ npm run test

> @mdn/browser-compat-data@3.1.1 test
> npm run mocha && npm run lint


> @mdn/browser-compat-data@3.1.1 mocha
> mocha "test/**.test.js"



  utils
    √ `escapeInvisibles()` works correctly


  1 passing (3ms)


> @mdn/browser-compat-data@3.1.1 lint
> node test/lint
> [rest of output]
```

### After:
```
$ npm run test

> @mdn/browser-compat-data@3.1.1 test /home/runner/work/browser-compat-data/browser-compat-data
> npm run mocha && npm run lint


> @mdn/browser-compat-data@3.1.1 mocha /home/runner/work/browser-compat-data/browser-compat-data
> mocha "test/**.test.js"



  compare-features script
    ✓ `compareFeatures()` works correctly

  utils
    ✓ `escapeInvisibles()` works correctly


  2 passing (7ms)


> @mdn/browser-compat-data@3.1.1 lint /home/runner/work/browser-compat-data/browser-compat-data
> node test/lint
> [rest of output]
```

A checklist to help your pull request get merged faster:
- [ ] Summarize your changes
- [ ] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [ ] Data: if you tested something, describe how you tested with details like browser and version
- [ ] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [ ] Link to related issues or pull requests, if any

### CODE REVIEW

**Comment by bershanskiy** [comment]
@ddbeck Is this PR something you would be interested in reviewing eventually or was `testCompareFeatures` intentionally placed into `test/lint.js`?

**Comment by ddbeck** [comment]
> Is this PR something you would be interested in reviewing eventually

Yes, I would like to review this! It's just… tough to find the time to do so. I'll see what I can do though.

**Comment by bershanskiy** [comment]
> Yes, I would like to review this! It's just… tough to find the time to do so. I'll see what I can do though.

Of course, take your time. I'm just checking in to make sure this PR is still relevant.

File: test/compare-features.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    if (errors) {
+      console.error(chalk`{red compareFeatures() – {bold 1} error:}`);
+      console.error(chalk`{red   → Actual and expected orders do not match}`);
+      console.error(chalk`{yellow     Actual: {bold ${actual}}}`);
+      console.error(chalk`{green     Expected: {bold ${expected}}}`);
+      return true;
+    }
+    return false;
+  });
```
Comment (queengooborg) [inline test/compare-features.test.js:64]: Like in the other PR, could we opt for `assert.deepStrictEqual(actual, expected)` instead?

**Review by queengooborg — CHANGES_REQUESTED** [review]
LGTM, just a little nit:

File: test/compare-features.test.js (hunk 1 above)
Comment (bershanskiy) [reply test/compare-features.test.js]: Done!

**Review by queengooborg — APPROVED** [review]
LGTM, thank you!

**Pull request merged**